src/amoc_config.erl

```diff
@@ -5,41 +5,103 @@
 -module(amoc_config).
 
 -export([get/1,
-         get/2,
-         fetch/1]).
+         get/2]).
+
+-export([parse_scenario_settings/1,
+         get_scenario_parameter/2]).
+
+-export_type([scenario_configuration/0, settings/0]).
+
+-type name() :: atom().
+-type env_var() :: atom().
+-type value() :: any().
+
+-type one_of() :: [any()].
+-type func() :: fun((any()) -> boolean()).
+-type validation() :: func() | one_of() | bitstring | nonnegative_integer | positive_integer | none.
+-type reason() :: atom().
+
+-type parameter_configuration() :: {name(), env_var(), DefValue :: value(), validation()}.
+-type scenario_configuration() :: [parameter_configuration()].
+
+-type settings() :: #{name() => value()}.
+-type invalid_settings() :: {invalid_settings, [{name(), value(), reason()}]}.
 
 %% ------------------------------------------------------------------
 %% API
 %% ------------------------------------------------------------------
--spec get(atom()) -> any().
+-spec get(name()) -> any().
 get(Name) ->
     get(Name, undefined).
 
--spec get(atom(), any()) -> any().
+-spec get(name(), any()) -> any().
 get(Name, Default) ->
     case os:getenv("AMOC_" ++ erlang:atom_to_list(Name)) of
         false ->
             application:get_env(amoc, Name, Default);
         Value ->
             parse_value(Value)
-    end.       
+    end.
 
--spec fetch(atom()) -> any().
-fetch(Name) ->
```

Comment:
are you sure that none of the scenarios is using it?

Reply:
I checked, and non in the repo uses this function. If there is other scenario outside the repo, I don't know.